test/ethregistrar/TestEthRegistrarController.js

```diff
@@ -1,185 +1,727 @@
+          registrantAccount,
+          secret,
+          NULL_ADDRESS,
+          [],
+          false,
+          0,
+          {
+            value: 28 * DAYS,
+          }
+        )
+      )
+    })
 
     it('should reject for expired commitments', async () => {
-        await controller.commit(await controller.makeCommitment("newname2", registrantAccount, secret));
-
-        await evm.advanceTime((await controller.maxCommitmentAge()).toNumber() + 1);
-        var balanceBefore = await web3.eth.getBalance(controller.address);
-        await exceptions.expectFailure(controller.register("newname2", registrantAccount, 28 * DAYS, secret, {value: 28 * DAYS}));
-    });
+      await controller.commit(
+        await controller.makeCommitment(
+          'newname2',
+          registrantAccount,
+          secret,
+          NULL_ADDRESS,
+          [],
+          false,
+          0
+        )
+      )
+
+      await evm.advanceTime(
+        (await controller.maxCommitmentAge()).toNumber() + 1
+      )
+      var balanceBefore = await web3.eth.getBalance(controller.address)
+      await exceptions.expectFailure(
+        controller.register(
+          'newname2',
+          registrantAccount,
+          secret,
+          NULL_ADDRESS,
+          [],
+          false,
+          0,
+          {
+            value: 28 * DAYS,
+          }
+        )
+      )
+    })
 
     it('should allow anyone to renew a name', async () => {
-        var expires = await baseRegistrar.nameExpires(sha3("newname"));
-        var balanceBefore = await web3.eth.getBalance(controller.address);
-        await controller.renew("newname", 86400, {value: 86400 + 1});
-        var newExpires = await baseRegistrar.nameExpires(sha3("newname"));
-        assert.equal(newExpires.toNumber() - expires.toNumber(), 86400);
-        assert.equal((await web3.eth.getBalance(controller.address)) - balanceBefore, 86400);
-    });
+      var expires = await baseRegistrar.nameExpires(sha3('newname'))
+      var balanceBefore = await web3.eth.getBalance(controller.address)
+      const targetDuration = 86400
+      const [price] = await controller.rentPrice(
+        sha3('newname'),
+        targetDuration
+      )
+      const [duration] = await controller.rentDuration(sha3('newname'), price)
+      expect(duration.toNumber()).to.equal(targetDuration)
+      await controller.renew('newname', { value: price })
+      var newExpires = await baseRegistrar.nameExpires(sha3('newname'))
+      assert.equal(newExpires.toNumber() - expires.toNumber(), 86400)
+      assert.equal(
+        (await web3.eth.getBalance(controller.address)) - balanceBefore,
+        86400
+      )
+    })
 
     it('should require sufficient value for a renewal', async () => {
-        await exceptions.expectFailure(controller.renew("name", 86400));
-    });
+      await exceptions.expectFailure(controller.renew('name'))
+    })
 
     it('should allow the registrar owner to withdraw funds', async () => {
```

Comment:
probably more correct to say "should allow anyone to withdraw funds and transfer to the registrar owner"